cypress/e2e/SubdomainNavigation.cy.ts

```diff
@@ -0,0 +1,221 @@
+
+		cy.visit(`//0.${subdomainZna}/nfts`);
+
+		// assert url
+		cy.assertUrl('contain', subdomainZna);
+		cy.assertUrl('not.contain', rootDomainZna);
+
+		// assert address bar anchor segments
+		cy.get('.main__header')
+			.find('a')
+			.should('have.length', 2)
+			.each((a, index) => {
+				// assert href attribute and anchor text for each anchor
+				const expectedHref =
+					index === 0 ? `/0.${rootDomainZna}/nfts` : `/0.${subdomainZna}/nfts`;
+				const expectedText = index === 0 ? 'wilder' : 'snowflake';
+
+				cy.wrap(a)
+					.should('have.attr', 'href', expectedHref)
+					.should('have.text', expectedText);
+			});
+
+		// assert expected elements
+		cy.assertElementIsVisible(
+			SubdomainViewSelector.VIEW_SUBDOMAINS_BANNER_CONTAINER,
+		);
+		cy.assertElementIsVisible(
+			SubdomainViewSelector.VIEW_SUBDOMAINS_DETAILS_CARD_CONTAINER,
+		).and('contain', 'Winter Time');
+	});
+
+	it('can view grid view and go to corret domain', () => {
+		cy.visitRootDomain();
+
+		// assert expected section element
+		cy.assertElementIsVisible(SubdomainViewSelector.VIEW_SUBDOMAINS_SECTION);
+
+		// find toggle view radio group
+		cy.findByRole('group').should('be.visible');
+
+		// find toggle view radio buttons and assert attributes
+		cy.findAllByRole('radio')
+			.should('be.visible')
+			.should('have.length', 2)
+			.each(($radio, index) => {
+				if (index === 0) {
+					cy.wrap($radio)
+						.should('have.attr', 'data-state', 'off')
+						.should('have.attr', 'aria-checked', 'false');
+				} else if (index === 1) {
+					cy.wrap($radio)
+						.should('have.attr', 'data-state', 'on')
+						.should('have.attr', 'aria-checked', 'true');
+				}
+			});
+
+		// assert grid view
+		cy.get('[class*="Grid"]').should('be.visible');
+
+		// find and click sudomain grid card
+		cy.findByText(`0://${subdomainZna}`).should('exist').click();
+
+		// assert url
+		cy.assertUrl('contain', subdomainZna);
+	});
+
+	it('can view list view and go to corret domain', () => {
+		cy.visitRootDomain();
+
+		// assert expected section element
+		cy.assertElementIsVisible(SubdomainViewSelector.VIEW_SUBDOMAINS_SECTION);
+
+		// find toggle view radio group
+		cy.findByRole('group').should('be.visible');
+
+		// find list view radio button and click
+		cy.findAllByRole('radio').first().click();
+
+		// find toggle view radio buttons and assert attributes
+		cy.findAllByRole('radio')
+			.should('have.length', 2)
+			.each(($radio, index) => {
+				if (index === 0) {
+					cy.wrap($radio)
+						.should('have.attr', 'data-state', 'on')
+						.should('have.attr', 'aria-checked', 'true');
+				} else if (index === 1) {
+					cy.wrap($radio)
+						.should('have.attr', 'data-state', 'off')
+						.should('have.attr', 'aria-checked', 'false');
+				}
+			});
+
+		// assert list view
+		cy.assertElementIsVisible('table');
+
+		// find and click sudomain grid card
+		cy.findByText(`0://${subdomainZna}`).should('exist').click();
```

Comment:
Possible to do the logic `a table with the text "0://subdomainZna" exists`? This way, if we add another table to the page for whatever reason, this test will still be accurate

Reply:
Sure can 💪 :

```
		// assert list view
		cy.assertElementIsVisible('table').and('contain', `0://${subdomainZna}`);

		// find and click sudomain grid card
		cy.findByText(`0://${subdomainZna}`).click();
```